tests/test_modeling_physical_validation.py

```diff
@@ -0,0 +1,161 @@
+
+def test_rho():
+    '''
+    Test physical values of rho
+    '''
+    rho = clmm.get_3d_density(R3D, **RHO_PARAMS)
+    tst.assert_allclose(NC_PROF['rho'], rho*G_PHYSCONST_CORRECTION, 1e-11)
+
+def test_sigma():
+    '''
+    Test physical values of sigma
+    '''
+    tst.assert_allclose(NC_PROF['Sigma'], G_PHYSCONST_CORRECTION*\
+                        clmm.predict_surface_density(R3D, **SIGMA_PARAMS), 1e-9)
+
+def test_delta_sigma():
+    '''
+    Test physical values of delta_sigma
+    '''
+    tst.assert_allclose(NC_PROF['DeltaSigma'], G_PHYSCONST_CORRECTION*\
+                        clmm.predict_excess_surface_density(R3D, **SIGMA_PARAMS), 1e-8)
+
+def test_get_da():
+    '''
+    Test physical values of Da
+    '''
+    dl_clmm = clmm.modeling.get_angular_diameter_distance_a(COSMO_CCL,
+                                                            TEST_CASE['aexp_cluster'])
+    ds_clmm = clmm.modeling.get_angular_diameter_distance_a(COSMO_CCL,
+                                                            TEST_CASE['aexp_source'])
+    dsl_clmm = clmm.modeling.get_angular_diameter_distance_a(COSMO_CCL,
+                                                             TEST_CASE['aexp_source'],
+                                                             TEST_CASE['aexp_cluster'])
+    tst.assert_allclose(TEST_CASE['dl'], dl_clmm, 1e-10)
+    tst.assert_allclose(TEST_CASE['ds'], ds_clmm, 1e-10)
+    tst.assert_allclose(TEST_CASE['dsl'], dsl_clmm, 1e-10)
+
+def test_sigmac():
+    '''
+    Test physical values of Sigmac
+    '''
+    tst.assert_allclose(TEST_CASE['nc_Sigmac'], SIGMAC_PHYSCONST_CORRECTION*\
+                        clmm.get_critical_surface_density(COSMO_CCL,
+                                                          z_cluster=TEST_CASE['z_cluster'],
+                                                          z_source=TEST_CASE['z_source']),
+                        1e-8)
+
+def test_gammat():
+    '''
+    Test physical values of gammat
+    '''
+    gammat = clmm.predict_tangential_shear(R3D, **GAMMA_PARAMS)
+    tst.assert_allclose(NC_PROF['gammat'], gammat/SIGMAC_PHYSCONST_CORRECTION, 1e-8)
+
+def test_kappa():
+    '''
+    Test physical values of kappa
+    '''
+    kappa = clmm.predict_convergence(R3D, **GAMMA_PARAMS)
+    tst.assert_allclose(NC_PROF['kappa'], kappa/SIGMAC_PHYSCONST_CORRECTION, 1e-8)
+
+def test_gt():
+    '''
+    Test physical values of gt
+    '''
+    gammat = clmm.predict_tangential_shear(R3D, **GAMMA_PARAMS)
+    kappa = clmm.predict_convergence(R3D, **GAMMA_PARAMS)
+    tst.assert_allclose(NC_PROF['gt'], gammat/(SIGMAC_PHYSCONST_CORRECTION-kappa), 1e-6)
+
+# others: test that inputs are as expected, values from demos
+# positive values from sigma onwards
+if __name__ == '__main__':
+    test_physical_constants()
+    test_rho()
+    test_sigma()
+    test_get_da()
+    test_sigmac()
+    test_delta_sigma()
+    test_gammat()
+    test_kappa()
+    test_gt()
```

Comment:
Shouldn't have a main funciton like this.

Reply:
fixed